### PR TITLE
Initial CI

### PR DESCRIPTION
**Pull request by dalazx**
https://circleci.com/gh/neuromation/ml-recipe-object-detection/15

https://hub.docker.com/repository/docker/neuromation/ml-recipe-object-detection

### CODE REVIEW

**Review by atemate — APPROVED** [review]
:+1:

**Comment by dalazx** [comment]
related: https://github.com/neuromation/platform-pipelines/pull/11/files

File: Makefile [hunk 1]
```diff
@@ -71,9 +71,22 @@ setup: ### Setup remote environment
...
+ifdef __BAKE_SETUP
+	make __bake
+endif
 	$(NEURO) --network-timeout 300 job save $(SETUP_JOB) $(CUSTOM_ENV_NAME)
 	$(NEURO) kill $(SETUP_JOB)
 
+.PHONY: __bake
+__bake: upload-code upload-data upload-notebooks
+	echo "#!/usr/bin/env bash" > jupyter.sh
+	echo "jupyter notebook --no-browser --ip=0.0.0.0 --allow-root \
```
Comment (atemate) [inline Makefile:83]: considering this pr, let's get rid of this code?
https://github.com/neuromation/neuro-base-environment/pull/91

File: Makefile [hunk 2]
```diff
@@ -71,9 +71,22 @@ setup: ### Setup remote environment
...
+	    --NotebookApp.token= --notebook-dir=/project/notebooks" >> jupyter.sh
+	$(NEURO) cp jupyter.sh $(PROJECT_PATH_STORAGE)/jupyter.sh
+	$(NEURO) exec --no-tty --no-key-check $(SETUP_JOB) \
+	    "bash -c 'mkdir -p /project; cp -R -T $(PROJECT_PATH_ENV) /project'"
```
Comment (atemate) [inline Makefile:87]: and here also, let's not introduce this hack but rather make path '/project' within the job for all Template projects?

**Review by atemate — COMMENTED** [review]
a few comments

File: Makefile (hunk 1 above)
Comment (dalazx) [reply Makefile:83]: no, that one is not merged nor released.

File: Makefile (hunk 2 above)
Comment (dalazx) [reply Makefile:87]: this could be `/whatever`. I don't think this is related to the scope of this PR.

File: Makefile (hunk 1 above)
Comment (atemate) [reply Makefile:83]: it's green, so we can do both right today

Comment (dalazx) [reply Makefile:83]: I'd rather do this in a separate PR.

File: Makefile (hunk 2 above)
Comment (atemate) [reply Makefile:87]: I meant to change it in Tepmlate https://github.com/neuromation/cookiecutter-neuro-project/issues/353
Anyways we are going to release Template v1.4 soon, there's just [Windows integration](https://github.com/neuromation/cookiecutter-neuro-project/pull/350) pending for review

**Comment by atemate** [comment]
I think, in order not to populate hacks over all recipes, we better fix base env + template (it's matter of days).

**Comment by dalazx** [comment]
@ayushkovskiy please elaborate why this's a hack? :) I do not see another way around. you still need to bake the files. no storage assumptions. no project.

**Comment by atemate** [comment]
> please elaborate why this's a hack?

Now what `__bake`does is it creates the shortcut for running jupyter (is gonna be fixed in https://github.com/neuromation/neuro-base-environment/pull/91) and moves project's files from `$(PROJECT_PATH_ENV)` to `/project` (whereas I'd like to get rid of this variable and use constant value `/project`, see https://github.com/neuromation/cookiecutter-neuro-project/issues/353). 

Regarding the second issue, @mariyadavydova might be against but I hope she'll agree that constant path is more convenient to work with.

**Comment by dalazx** [comment]
thanks @ayushkovskiy those are all good points, but I still do not understand my you think it is a hack. the main thing there is that we copy the files from the storage volume to the container FS and then `save` the files right into the image. there's no way around it for now.

**Comment by dalazx** [comment]
once the proposed fixed are in place and released, we could just slightly patch `Makefile`. merging.

**Comment by atemate** [comment]
> but I still do not understand my you think it is a hack. the main thing there is that we copy the files from the storage volume to the container FS and then save the files right into the image

Yes, thank you for the explanation. I didn't realize that the main idea of these movements were to put data inside the image. Thanks for your explanations!

**Pull request merged**